src/trace/step-function-service.spec.ts

```diff
@@ -268,36 +269,44 @@ describe("StepFunctionContextService", () => {
       const times = 20;
       for (let i = 0; i < times; i++) {
         for (let j = i + 1; j < times; j++) {
-          const hash1 = instance["deterministicMd5HashInBinary"](i.toString());
-          const hash2 = instance["deterministicMd5HashInBinary"](j.toString());
+          const hash1 = instance["deterministicSha256Hash"](i.toString(), _128_BITS);
+          const hash2 = instance["deterministicSha256Hash"](j.toString(), _128_BITS);
           expect(hash1).not.toMatch(hash2);
         }
       }
     });
   });
 
-  describe("hexToBinary", () => {
+  describe("numberToBinaryString", () => {
     const instance = StepFunctionContextService.instance();
     it.each([
-      ["0", "0000"],
-      ["1", "0001"],
-      ["2", "0010"],
-      ["3", "0011"],
-      ["4", "0100"],
-      ["5", "0101"],
-      ["6", "0110"],
-      ["7", "0111"],
-      ["8", "1000"],
-      ["9", "1001"],
-      ["a", "1010"],
-      ["b", "1011"],
-      ["c", "1100"],
-      ["d", "1101"],
-      ["e", "1110"],
-      ["f", "1111"],
+      [0, "00000000"],
+      [1, "00000001"],
+      [2, "00000010"],
+      [3, "00000011"],
+      [4, "00000100"],
     ])("returns the right binary number for %s => %s", (hex, expected) => {
-      const binary = instance["hexToBinary"](hex);
+      const binary = instance["numberToBinaryString"](hex);
       expect(binary).toBe(expected);
     });
   });
+
+  describe("test 64 bits deterministicMd5HashToBigIntString for span id", () => {
+    const instance = StepFunctionContextService.instance();
+    it("first test of #1", () => {
+      const actual = instance["deterministicSha256HashToBigIntString"](
+        "arn:aws:states:sa-east-1:425362996713:stateMachine:MyStateMachine-b276uka1j#lambda#1",
+        _64_BITS,
+      );
+      expect(actual).toEqual("3711631873188331089");
+    });
+
+    it("test same hashing number is generated as logs-backend for execution id # state name # entered time", () => {
+      const actual = instance["deterministicSha256HashToBigIntString"](
+        "arn:aws:states:sa-east-1:425362996713:stateMachine:MyStateMachine-b276uka1j#lambda#2",
+        _64_BITS,
+      );
+      expect(actual).toEqual("5759173372325510050");
```

Comment:
this matches logs-backend's hash result
https://github.com/DataDog/logs-backend/pull/71616/files#diff-820e0a1dd4f0f97815ac54f993c08b3c44c67dd26a312968c214f16ab73494c0R20-R21